### PR TITLE
Issue 122 venue post and get

### PR DESCRIPTION
**Pull request by OughtNaut**
I thought i had made a pull request for this already, but I can't seem to find evidence of it being denied. If I'm doing something wrong please let me know.

### CODE REVIEW

File: backend/internal/server/apis/stats/venues.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		return responder.InternalServerError(c)
+	}
+
+	if record == nil {
+		return responder.BadRequest(c, "Could not post venue into database")
+	}
+	return responder.Ok(c)
+}
+
+// todo finish getVenuesHandler
```
Comment (tannerwheeler) [inline backend/internal/server/apis/stats/venues.go]: @OughtNaut What do you need to finish here? Is this TODO comment okay now? If not, let's create a new issue if this is future work and remove the comment on line `42`.

File: static/oas/v1/stats/venues.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+paths:
+  venues:
```
Comment (tannerwheeler) [inline static/oas/v1/stats/venues.yml:2]: If I am following past Pull Requests correctly, this will need to be /venues.

**Review by tannerwheeler — CHANGES_REQUESTED** [review]
See the comment about the TODO and the /

File: backend/internal/server/server.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ func Run() {
 
 	apis.SetupRoutes(app)
 
-	app.Static("/", "/app/static")
+	app.Static("/", "/powerplay/static")
```
Comment (ryanbarlow1) [inline backend/internal/server/server.go]: Pretty sure this is a big enough change it needs to be in its own PR

File: static/docs/index.html [hunk 4]
```diff
@@ -8,7 +8,7 @@
     <rapi-doc 
         spec-url = "/oas/v1/openapi.yml"
         show-header	="false"
-        server-url = "http://localhost:9001/api/v1" 
+        server-url = "http://localhost:9002/api/v1"
```
Comment (ryanbarlow1) [inline static/docs/index.html:11]: Same, this should be in its own PR with the server.go change, but great catch

File: static/oas/v1/stats/venues.yml (hunk 2 above)
Comment (ryanbarlow1) [reply static/oas/v1/stats/venues.yml:2]: It can be either one, but `/venues` needs to be used if there's endpoints for both `/venues` (for Get All and Post) and `/venues:id` (for Get By ID and PUT)

However since this endpoint only has a Get All and Post, either one works

File: backend/internal/server/apis/stats/venues.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	db := db.GetSession(c)
+	record, err := db.SaveVenue(venueRequest)
+	
+	if err != nil {
+		log.WithErr(err).Error("Failed to save venue request")
+		return responder.InternalServerError(c)
+	}
+
+	if record == nil {
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/stats/venues.go]: Interesting way of doing this with passing the record back. I'm not sure if record will ever be nil though since line 24 above would've caught it.

**Review by ryanbarlow1 — CHANGES_REQUESTED** [review]

File: static/docs/index.html (hunk 4 above)
Comment (tannerwheeler) [reply static/docs/index.html:11]: This was already changed by Dr. Christensen. You have merge conflicts and this is one of them. You are okay to push this. It has already been changed in the main project and matches your change.

File: backend/internal/server/server.go (hunk 3 above)
Comment (tannerwheeler) [reply backend/internal/server/server.go]: This was already changed by Dr. Christensen. You are okay to push this. It has already been changed in the main project and matches your change.

**Comment by tannerwheeler** [comment]
The RapiDocs, at `localhost:9002/docs`, is working for me.

File: backend/internal/server/apis/stats/venues.go (hunk 5 above)
Comment (OughtNaut) [reply backend/internal/server/apis/stats/venues.go]: @ryanbarlow1 I removed this check in my most recent commit, let me know if this looks good now.

File: backend/internal/server/apis/stats/venues.go (hunk 1 above)
Comment (OughtNaut) [reply backend/internal/server/apis/stats/venues.go]: @tannerwheeler  sorry I missed this, I have removed this in the most recent commit.

**Review by ryanbarlow1 — APPROVED** [review]

**Review by tannerwheeler — APPROVED** [review]

File: static/oas/v1/stats/venues.yml (hunk 2 above)
Comment (ryanbarlow1) [reply static/oas/v1/stats/venues.yml:2]: I got the syntax wrong on this, Get By ID and PUT should be under `/venues/{venueId}`
